apps/client/sttr_build/parser.js

```diff
@@ -52,165 +75,288 @@ const getAutofillResolverKey = (questionText) => {
+        result[shortKey] = (result[shortKey] || []).concat(
+          requiredElement.attribs.href
+        );
+        return result;
+      },
+      {}
+    );
 
-  /**
-   * Get decisions configuration
-   *
-   * @returns {any} a configuration object for decisions
-   */
-  get decisions() {
-    return this.xmlDecisions.reduce((xmlDecisions, xmlDecision) => {
-      const res = xmlDecision["dmn:informationRequirement"].reduce(
-        (acc, ir) => {
-          const key = Object.keys(ir)[0]; // get the tagName
-          const shortKey = `${key.split(":")[1]}s`;
-          const href = ir[key][0]["@_href"];
-
-          const result = acc;
-          result[shortKey] = (result[shortKey] || []).concat(href);
-          return result;
-        },
-        {}
-      );
+    const table = find(xmlDecision.children, "dmn:decisionTable");
+    debug("table", log(table));
 
-      const table = xmlDecision["dmn:decisionTable"][0];
-      const rules = table["dmn:rule"].reduce((acc, rule) => {
-        const outputEntry = get(rule, "dmn:outputEntry.0");
-        const description = get(
-          outputEntry,
-          "dmn:extensionElements.0.content:conclusieToelichting.0.content:toelichting"
+    const rules = filt(table.children, "dmn:rule").reduce((rules, rule) => {
+      debug("rule", log(rule));
+      const outputEntry = filt(rule.children, "dmn:outputEntry")[0];
+      debug("outputEntry", log(outputEntry));
+
+      const extensionElements = find(
+        outputEntry.children,
+        "dmn:extensionElements"
+      );
+      let descriptionText = undefined;
+      if (extensionElements) {
+        debug("extensionElements", log(extensionElements));
+        const conclusionDescription = find(
+          extensionElements.children,
+          "content:conclusieToelichting"
         );
-        acc.push({
-          inputs: rule["dmn:inputEntry"].reduce((inputEntry, ie) => {
-            inputEntry.push(ie["dmn:text"]);
-            return inputEntry;
-          }, []),
-          output: outputEntry["dmn:text"],
-          description,
-        });
-        return acc;
-      }, []);
-      res.decisionTable = {
-        rules,
+        debug("conclusionDescription", log(conclusionDescription));
+        const description = find(
+          conclusionDescription.children,
+          "content:toelichting"
+        ).children[0].children;
+        debug("ruledescription", log(description));
+        descriptionText = description[0].data;
+      }
+
+      const output = find(outputEntry.children, "dmn:text");
+      debug("output", log(output));
+      rules.push({
+        description: descriptionText,
+        inputs: filt(rule.children, "dmn:inputEntry").reduce(
+          (inputEntries, inputEntry) => {
+            const text = find(inputEntry.children, "dmn:text").children[0].data;
+            debug("inputtext", text);
+            const map = {
+              false: false,
+              true: true,
```

Comment:
It this really needed? It looks awkward 🌫️

Reply:
will fix.